groups/validate.go

```diff
@@ -104,196 +134,232 @@ func TestMergedGroupsConfig(t *testing.T) {
 	}
 
 	if !containsMergedConfig {
-		t.Errorf("Final GroupsConfig does not have merged configs from all groups.yaml files")
+		fmt.Printf("Final GroupsConfig does not have merged configs from all groups.yaml files")
+		os.Exit(1)
 	}
 	if n := len(dups); n > 0 {
-		t.Errorf("%d duplicate groups: %s", n, strings.Join(dups.List(), ", "))
+		fmt.Printf("%d duplicate groups: %s", n, strings.Join(dups.List(), ", "))
+		os.Exit(1)
 	}
 }
 
-// TestStagingEmailLength tests that the number of characters in the
+// VerifyStagingEmailLength tests that the number of characters in the
 // project name in emails used for staging repos does not exceed 18.
 //
 // This validation is needed because gcloud allows PROJECT_IDs of length
 // between 6 and 30. So after discounting the "k8s-staging" prefix,
 // we are left with 18 chars for the project name.
-func TestStagingEmailLength(t *testing.T) {
+func VerifyStagingEmailLength(config Config) {
```

Comment:
Do you want to pass `groupsConfig.Groups` as a parameter here, too, so that it's not referencing the shared global. (Mostly for testing convenience and to make this a pure function.)